numpy/linalg/linalg.py

```diff
@@ -2201,18 +2201,7 @@ def norm(x, ord=None, axis=None, keepdims=False):
                 ord + 1
             except TypeError:
                 raise ValueError("Invalid norm order for vectors.")
-            if x.dtype.type is longdouble:
-                # Convert to a float type, so integer arrays give
-                # float results.  Don't apply asfarray to longdouble arrays,
-                # because it will downcast to float64.
-                absx = abs(x)
-            else:
-                absx = x if isComplexType(x.dtype.type) else asfarray(x)
-                if absx.dtype is x.dtype:
-                    absx = abs(absx)
-                else:
-                    # if the type changed, we can safely overwrite absx
-                    abs(absx, out=absx)
+            absx = abs(x)
```

Comment:
This is not quite the same as `asfarray` casts to float64 type but `abs` does not. In particular this will affect float16 type.  OTOH, float16 is preserved by the other norms. Hmm... IIRC, this function has had various type problems over the years. So the change may be justified, but might be worth something in the release notes.